pkg/apis/machine/types.go

```diff
@@ -1204,16 +1204,19 @@ type MachineClass struct {
 	metav1.TypeMeta
 	// +optional
 	metav1.ObjectMeta
-	// Provider-specific configuration to use during node creation.
-	ProviderSpec runtime.RawExtension
-	// SecretRef stores the necessary secrets such as credentials or userdata.
-	SecretRef *corev1.SecretReference
 	// CredentialsSecretRef can optionally store the credentials (in this case the SecretRef does not need to store them).
 	// This might be useful if multiple machine classes with the same credentials but different user-datas are used.
 	CredentialsSecretRef *corev1.SecretReference
+	// ExpectedNodeAttributes contains subfields the track all scale from zero resources
+	// +optional
+	ExpectedNodeAttributes *ExpectedNodeAttributes `json:"expectedNodeAttributes,omitempty"`
```

Comment:
```suggestion
	ExpectedNodeAttributes *ExpectedNodeAttributes
```
The `+optional` and `json` strings are not required in the Main API. It is only required in the versioned APIs. 
Refer gardener APIs for more insights in future.
- https://github.com/gardener/gardener/blob/master/pkg/apis/core/types_shoot.go
- https://github.com/gardener/gardener/blob/master/pkg/apis/core/v1alpha1/types_shoot.go

Reply:
Oh, okay! Got it.